script.js

```diff
@@ -81,17 +88,14 @@ function evaluateNow() {
     }
     updateDisplay(fourDPIfNonInteger(ans)); // 4 dp display if not integer
     operatorArray.pop(); // This function always uses last element, so pop out after using
-    numberArray = [ans]; // Store exact value for carry-on calculations
+    numberArray.push(ans); // Store exact value for carry-on calculations
```

Comment:
Don't clear. Want to keep unused number. The previous clearing method works because rebuilding the whole array, but that method has been replaced by .pop().